receiver/sqlserverreceiver/scraper_test.go

```diff
@@ -31,7 +32,7 @@ func TestSqlServerScraper(t *testing.T) {
 	factory := NewFactory()
 	cfg := factory.CreateDefaultConfig().(*Config)
 	logger, obsLogs := observer.New(zap.WarnLevel)
-	s := newSqlServerScraper(zap.New(logger), cfg)
+	s := newSqlServerScraper(componenttest.NewNopReceiverCreateSettings(), cfg)
```

Comment:
This will likely fail. You need to pass the `logger` with the settings

Reply:
Fixed https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/10137/commits/2dda3a82a6aad66fc56a69b7aeddb942cf92a90c